pkg/auth/auth_config_service.go

```diff
@@ -0,0 +1,49 @@
+package auth
+
+// Config gets the AuthConfig from the service
+func (s *GenericAuthConfigService) Config() *AuthConfig {
+	if s.config == nil {
+		s.config = &AuthConfig{}
+	}
+	return s.config
+}
+
+// SetConfig sets the AuthConfig object
+func (s *GenericAuthConfigService) SetConfig(c *AuthConfig) {
+	s.config = c
+}
+
+// SaveUserAuth saves the given user auth for the server url
+func (s *GenericAuthConfigService) SaveUserAuth(url string, userAuth *UserAuth) error {
+	config := s.config
+	config.SetUserAuth(url, userAuth)
+	user := userAuth.Username
+	if user != "" {
+		config.DefaultUsername = user
+	}
+
+	// Set Pipeline user once only.
+	if config.PipeLineUsername == "" {
+		config.PipeLineUsername = user
+		config.PipeLineServer = url
+	}
+
+	config.CurrentServer = url
+	return s.saver.SaveConfig(s.config)
```

Comment:
NIT: This call looks quite weird to me. `s` already has everything to do the job (which is `s.saver` and `s.config`), so something like `s.SaveConfig()` seems more natural.

Reply:
it looks weird to me too, but adding a wrapper method doesn't appear to be the go way